### ISSUE TITLE
coco segmentation data for polygon should be [polygon,] not polygon

### ISSUE THREAD

**Issue by yance-dev** [roboflow/supervision#205]
thanks for this great project, but I found that supervision export coco as :

```
"segmentation":  polygon,
```
https://github.com/roboflow/supervision/blob/20954a299660a1f227a772a0a7cd253761ddb9da/supervision/dataset/formats/coco.py#L127

but coco official [data format](https://cocodataset.org/#format-data:~:text=%22segmentation%22,RLE%20or%20%5Bpolygon%5D%2C) is :
```
"segmentation":  [polygon],
```
even only one polygon, it should be a list of polygons.

the third party like fifityone may not support this format.

that's all, thanks again.

**Comment by SkalskiP** [comment]
Hi, @yance-dev 👋🏻! Thanks a lot for your interest in `supervision`. Let me take a look and confirm if the problem exists. 

Would you like to create a PR with fix?

**Comment by SkalskiP** [comment]
Hi, @yance-dev. I just did some tests and it looks like everything work just fine on our side. Did you face any issues?

**Comment by onuralpszr** [comment]
It does work because polygon is a "list" checkout here as well

https://github.com/roboflow/supervision/blob/20954a299660a1f227a772a0a7cd253761ddb9da/supervision/dataset/formats/coco.py#L111C12-L111C12

**Comment by SkalskiP** [comment]
It looks like a false alarm this time. @yance-dev I'm closing the issue for now, but feel free to reopen it in the future.

**Comment by yance-dev** [comment]
> It does work because polygon is already a "list" checkout here as well
> 
> https://github.com/roboflow/supervision/blob/20954a299660a1f227a772a0a7cd253761ddb9da/supervision/dataset/formats/coco.py#L111
> 
> https://github.com/roboflow/supervision/blob/20954a299660a1f227a772a0a7cd253761ddb9da/supervision/dataset/formats/coco.py#L113

polygon is a list. but coco `"segmentation":` should be  polygons which is dyadic array like `[[x1,y1,x2,y2,x3,y3,x4,y4],[x1,y1,x2,y2,x3,y3,x4,y4]]`

https://github.com/roboflow/supervision/blob/20954a299660a1f227a772a0a7cd253761ddb9da/supervision/dataset/formats/coco.py#L119C31-L119C31
 ```
                approximate_mask_with_polygons(
                    mask=mask,
                    min_image_area_percentage=min_image_area_percentage,
                    max_image_area_percentage=max_image_area_percentage,
                    approximation_percentage=approximation_percentage,
                )[0].flatten()
```
after this , a list of polygons return  the first one polygon, and flatten it to a list of points

and list(polygon) not change it to a list of polygons, but a list of points

**Comment by SkalskiP** [comment]
@yance-dev `polygon` is a list of lists of points. Single `approximate_mask_with_polygons` returns a single polygon. Because we run it in the list comprehension, you get a list of lists. As the result we get exactly this format `[[x1,y1,x2,y2,x3,y3,x4,y4],[x1,y1,x2,y2,x3,y3,x4,y4]]`.

Would be awesome if you could give us a reproducible example where code breaks.

**Comment by yance-dev** [comment]
Sorry for the delayed reply, I may repeat the experiment over the weekend, and if I can reproduce the issue, I will provide a corresponding test case. Thanks for everyone's reply

**Comment by SkalskiP** [comment]
@yance-dev, if you'd be able to provide us with a reproducible example I'd be super grateful. We will definitely work on appropriate fixes.

**Comment by yance-dev** [comment]
https://colab.research.google.com/drive/1DL-rbH6OCZ7k0VSN0hPaMh7PmJCdrtym?usp=sharing

In my tests, SV did make changes to the data structure, but sv seems to be compatible with this change. But as I mentioned above some third-party tools may report errors due to changes in data structures

![image](https://github.com/roboflow/supervision/assets/31528034/9ba138be-caa3-4487-b6c9-eead141d8fff)

original_coco.json:
```
{
    "info": {
        "contributor": "xxx",
        "date_created": "2023-07-21 09:20:23",
        "description": "",
        "url": "",
        "version": "1.0",
        "year": 2023
    },
    "licenses": [
        {
            "url": "http://creativecommons.org/licenses/by-nc-sa/2.0/",
            "id": 1,
            "name": "Attribution-NonCommercial-ShareAlike License"
        }
    ],
    "images": [

        {
            "id": 1664512,
            "width": 1080,
            "height": 607,
            "file_name": "241295174_802164147128543_3685831163417916581_n.jpg"
        }
    ],
    "annotations": [
        {
            "id": 2181749,
            "image_id": 1664512,
                       "segmentation": [
                [
                    1047,
                    412,
                    1040,
                    387,
                    1017,
                    359,
                    1010,
                    346,
                    996,
                    327,
                    991,
                    313,
                    986,
                    308,
                    981,
                    297,
                    975,
                    291,
                    968,
                    279,
                    962,
                    273,
                    950,
                    248,
                    935,
                    229,
                    916,
                    211,
                    893,
                    196,
                    876,
                    188,
                    867,
                    181,
                    798,
                    144,
                    777,
                    137,
                    759,
                    127,
                    728,
                    119,
                    707,
                    111,
                    681,
                    107,
                    664,
                    107,
                    622,
                    114,
                    589,
                    123,
                    558,
                    140,
                    545,
                    152,
                    538,
                    156,
                    526,
                    169,
                    517,
                    174,
                    505,
                    187,
                    490,
                    191,
                    486,
                    194,
                    479,
                    194,
                    480,
                    162,
                    475,
                    150,
                    458,
                    149,
                    437,
                    168,
                    427,
                    172,
                    415,
                    174,
                    406,
                    181,
                    355,
                    182,
                    309,
                    193,
                    293,
                    194,
                    284,
                    199,
                    279,
                    199,
                    262,
                    205,
                    247,
                    213,
                    228,
                    211,
                    217,
                    207,
                    202,
                    207,
                    175,
                    197,
                    155,
                    202,
                    154,
                    218,
                    160,
                    232,
                    172,
                    247,
                    178,
                    258,
                    186,
                    266,
                    204,
                    298,
                    210,
                    305,
                    213,
                    329,
                    220,
                    357,
                    235,
                    392,
                    244,
                    407,
                    249,
                    411,
                    249,
                    415,
                    239,
                    423,
                    225,
                    441,
                    223,
                    460,
                    227,
                    471,
                    234,
                    475,
                    249,
                    477,
                    270,
                    476,
                    280,
                    478,
                    313,
                    476,
                    332,
                    477,
                    335,
                    475,
                    338,
                    481,
                    336,
                    495,
                    337,
                    516,
                    339,
                    522,
                    344,
                    526,
                    346,
                    534,
                    370,
                    538,
                    419,
                    537,
                    436,
                    534,
                    441,
                    531,
                    528,
                    518,
                    547,
                    517,
                    557,
                    514,
                    583,
                    514,
                    600,
                    511,
                    610,
                    512,
                    616,
                    510,
                    715,
                    507,
                    725,
                    503,
                    788,
                    500,
                    800,
                    497,
                    815,
                    496,
                    837,
                    487,
                    865,
                    481,
                    968,
                    473,
                    990,
                    468,
                    998,
                    464,
                    1018,
                    461,
                    1034,
                    454,
                    1044,
                    444
                ]
            ],
            "area": 266507.0,
            "bbox": [
                154.0,
                107.0,
                894.0,
                432.0
            ],
            "iscrowd": 1,
            "category_id": 1,
            "meta": {
                "result_type": "ground_truth",
                "prompt": null
            }
        }
    ],
    "categories": [
        {
            "name": "cat",
            "supercategory": "cat",
            "id": 1
        }
    ]
}
```

export_coco.json:
```
{
   "info": {},
   "licenses": [
      {
         "id": 1,
         "url": "https://creativecommons.org/licenses/by/4.0/",
         "name": "CC BY 4.0"
      }
   ],
   "categories": [
      {
         "id": 0,
         "name": "cat",
         "supercategory": "common-objects"
      }
   ],
   "images": [
      {
         "id": 0,
         "license": 1,
         "file_name": "241295174_802164147128543_3685831163417916581_n.jpg",
         "height": 607,
         "width": 1080,
         "date_captured": "07/21/2023,11:28:21"
      }
   ],
   "annotations": [
      {
         "id": 0,
         "image_id": 0,
         "category_id": 1,
         "bbox": [
            154.0,
            107.0,
            894.0,
            432.0
         ],
         "area": 386208.0,
         "segmentation": [
            662,
            107,
            661,
            108,
            656,
            108,
            655,
            109,
            650,
            109,
            649,
            110,
            644,
            110,
            643,
            111,
            638,
            111,
            637,
            112,
            632,
            112,
            631,
            113,
            626,
            113,
            625,
            114,
            621,
            114,
            620,
            115,
            617,
            115,
            616,
            116,
            613,
            116,
            612,
            117,
            610,
            117,
            609,
            118,
            606,
            118,
            605,
            119,
            602,
            119,
            601,
            120,
            599,
            120,
            598,
            121,
            595,
            121,
            594,
            122,
            591,
            122,
            590,
            123,
            589,
            123,
            588,
            124,
            587,
            124,
            586,
            125,
            585,
            125,
            584,
            126,
            583,
            126,
            582,
            127,
            581,
            127,
            580,
            128,
            579,
            128,
            577,
            130,
            576,
            130,
            575,
            131,
            574,
            131,
            573,
            132,
            572,
            132,
            571,
            133,
            570,
            133,
            568,
            135,
            567,
            135,
            566,
            136,
            565,
            136,
            564,
            137,
            563,
            137,
            562,
            138,
            561,
            138,
            560,
            139,
            559,
            139,
            552,
            146,
            551,
            146,
            544,
            153,
            543,
            153,
            542,
            154,
            541,
            154,
            540,
            155,
            539,
            155,
            532,
            162,
            532,
            163,
            525,
            170,
            524,
            170,
            523,
            171,
            522,
            171,
            521,
            172,
            520,
            172,
            519,
            173,
            518,
            173,
            511,
            180,
            511,
            181,
            505,
            187,
            504,
            187,
            503,
            188,
            500,
            188,
            499,
            189,
            496,
            189,
            495,
            190,
            492,
            190,
            491,
            191,
            490,
            191,
            488,
            193,
            487,
            193,
            486,
            194,
            480,
            194,
            479,
            193,
            479,
            178,
            480,
            177,
            480,
            161,
            479,
            160,
            479,
            159,
            478,
            158,
            478,
            157,
            477,
            156,
            477,
            154,
            476,
            153,
            476,
            152,
            475,
            151,
            475,
            150,
            467,
            150,
            466,
            149,
            458,
            149,
            453,
            154,
            452,
            154,
            443,
            163,
            442,
            163,
            437,
            168,
            436,
            168,
            435,
            169,
            434,
            169,
            433,
            170,
            431,
            170,
            430,
            171,
            429,
            171,
            428,
            172,
            425,
            172,
            424,
            173,
            419,
            173,
            418,
            174,
            415,
            174,
            413,
            176,
            412,
            176,
            409,
            179,
            408,
            179,
            406,
            181,
            381,
            181,
            380,
            182,
            353,
            182,
            352,
            183,
            349,
            183,
            348,
            184,
            345,
            184,
            344,
            185,
            341,
            185,
            340,
            186,
            337,
            186,
            336,
            187,
            333,
            187,
            332,
            188,
            328,
            188,
            327,
            189,
            324,
            189,
            323,
            190,
            320,
            190,
            319,
            191,
            316,
            191,
            315,
            192,
            312,
            192,
            311,
            193,
            302,
            193,
            301,
            194,
            293,
            194,
            292,
            195,
            291,
            195,
            290,
            196,
            289,
            196,
            288,
            197,
            287,
            197,
            286,
            198,
            285,
            198,
            284,
            199,
            278,
            199,
            277,
            200,
            275,
            200,
            274,
            201,
            272,
            201,
            271,
            202,
            270,
            202,
            269,
            203,
            267,
            203,
            266,
            204,
            264,
            204,
            263,
            205,
            262,
            205,
            261,
            206,
            260,
            206,
            259,
            207,
            258,
            207,
            257,
            208,
            256,
            208,
            255,
            209,
            254,
            209,
            253,
            210,
            252,
            210,
            251,
            211,
            250,
            211,
            249,
            212,
            248,
            212,
            247,
            213,
            243,
            213,
            242,
            212,
            233,
            212,
            232,
            211,
            227,
            211,
            226,
            210,
            224,
            210,
            223,
            209,
            222,
            209,
            221,
            208,
            219,
            208,
            218,
            207,
            201,
            207,
            200,
            206,
            198,
            206,
            197,
            205,
            196,
            205,
            195,
            204,
            193,
            204,
            192,
            203,
            190,
            203,
            189,
            202,
            188,
            202,
            187,
            201,
            185,
            201,
            184,
            200,
            182,
            200,
            181,
            199,
            180,
            199,
            179,
            198,
            177,
            198,
            176,
            197,
            174,
            197,
            173,
            198,
            170,
            198,
            169,
            199,
            166,
            199,
            165,
            200,
            162,
            200,
            161,
            201,
            158,
            201,
            157,
            202,
            155,
            202,
            155,
            209,
            154,
            210,
            154,
            219,
            155,
            220,
            155,
            221,
            156,
            222,
            156,
            223,
            157,
            224,
            157,
            226,
            158,
            227,
            158,
            228,
            159,
            229,
            159,
            230,
            160,
            231,
            160,
            232,
            162,
            234,
            162,
            235,
            166,
            239,
            166,
            240,
            170,
            244,
            170,
            245,
            173,
            248,
            173,
            249,
            174,
            250,
            174,
            251,
            175,
            252,
            175,
            253,
            176,
            254,
            176,
            255,
            177,
            256,
            177,
            257,
            187,
            267,
            187,
            268,
            188,
            269,
            188,
            270,
            189,
            271,
            189,
            272,
            190,
            273,
            190,
            274,
            192,
            276,
            192,
            277,
            193,
            278,
            193,
            279,
            194,
            280,
            194,
            281,
            196,
            283,
            196,
            284,
            197,
            285,
            197,
            286,
            198,
            287,
            198,
            288,
            199,
            289,
            199,
            290,
            201,
            292,
            201,
            293,
            202,
            294,
            202,
            295,
            203,
            296,
            203,
            297,
            207,
            301,
            207,
            302,
            210,
            305,
            210,
            309,
            211,
            310,
            211,
            317,
            212,
            318,
            212,
            325,
            213,
            326,
            213,
            331,
            214,
            332,
            214,
            335,
            215,
            336,
            215,
            339,
            216,
            340,
            216,
            343,
            217,
            344,
            217,
            347,
            218,
            348,
            218,
            351,
            219,
            352,
            219,
            355,
            220,
            356,
            220,
            358,
            221,
            359,
            221,
            360,
            222,
            361,
            222,
            362,
            223,
            363,
            223,
            365,
            224,
            366,
            224,
            367,
            225,
            368,
            225,
            369,
            226,
            370,
            226,
            372,
            227,
            373,
            227,
            374,
            228,
            375,
            228,
            376,
            229,
            377,
            229,
            379,
            230,
            380,
            230,
            381,
            231,
            382,
            231,
            383,
            232,
            384,
            232,
            386,
            233,
            387,
            233,
            388,
            234,
            389,
            234,
            390,
            235,
            391,
            235,
            392,
            236,
            393,
            236,
            394,
            237,
            395,
            237,
            396,
            239,
            398,
            239,
            399,
            240,
            400,
            240,
            401,
            242,
            403,
            242,
            404,
            243,
            405,
            243,
            406,
            246,
            409,
            247,
            409,
            249,
            411,
            249,
            415,
            247,
            417,
            246,
            417,
            242,
            421,
            241,
            421,
            237,
            425,
            237,
            426,
            234,
            429,
            234,
            430,
            230,
            434,
            230,
            435,
            227,
            438,
            227,
            439,
            225,
            441,
            225,
            445,
            224,
            446,
            224,
            455,
            223,
            456,
            223,
            461,
            224,
            462,
            224,
            464,
            225,
            465,
            225,
            466,
            226,
            467,
            226,
            469,
            227,
            470,
            227,
            471,
            228,
            472,
            229,
            472,
            230,
            473,
            231,
            473,
            232,
            474,
            233,
            474,
            234,
            475,
            237,
            475,
            238,
            476,
            245,
            476,
            246,
            477,
            259,
            477,
            260,
            476,
            272,
            476,
            273,
            477,
            277,
            477,
            278,
            478,
            288,
            478,
            289,
            477,
            304,
            477,
            305,
            476,
            322,
            476,
            323,
            477,
            332,
            477,
            333,
            476,
            335,
            476,
            336,
            477,
            336,
            478,
            337,
            479,
            337,
            480,
            338,
            481,
            338,
            484,
            337,
            485,
            337,
            491,
            336,
            492,
            336,
            505,
            337,
            506,
            337,
            517,
            338,
            518,
            338,
            520,
            339,
            521,
            339,
            522,
            341,
            524,
            342,
            524,
            344,
            526,
            344,
            528,
            345,
            529,
            345,
            532,
            346,
            533,
            346,
            534,
            349,
            534,
            350,
            535,
            355,
            535,
            356,
            536,
            361,
            536,
            362,
            537,
            367,
            537,
            368,
            538,
            394,
            538,
            395,
            537,
            421,
            537,
            422,
            536,
            427,
            536,
            428,
            535,
            433,
            535,
            434,
            534,
            436,
            534,
            437,
            533,
            438,
            533,
            439,
            532,
            440,
            532,
            441,
            531,
            444,
            531,
            445,
            530,
            451,
            530,
            452,
            529,
            457,
            529,
            458,
            528,
            464,
            528,
            465,
            527,
            471,
            527,
            472,
            526,
            477,
            526,
            478,
            525,
            484,
            525,
            485,
            524,
            491,
            524,
            492,
            523,
            497,
            523,
            498,
            522,
            504,
            522,
            505,
            521,
            511,
            521,
            512,
            520,
            517,
            520,
            518,
            519,
            524,
            519,
            525,
            518,
            537,
            518,
            538,
            517,
            548,
            517,
            549,
            516,
            552,
            516,
            553,
            515,
            555,
            515,
            556,
            514,
            585,
            514,
            586,
            513,
            591,
            513,
            592,
            512,
            597,
            512,
            598,
            511,
            605,
            511,
            606,
            512,
            611,
            512,
            612,
            511,
            614,
            511,
            615,
            510,
            632,
            510,
            633,
            509,
            665,
            509,
            666,
            508,
            698,
            508,
            699,
            507,
            716,
            507,
            717,
            506,
            718,
            506,
            719,
            505,
            721,
            505,
            722,
            504,
            723,
            504,
            724,
            503,
            735,
            503,
            736,
            502,
            756,
            502,
            757,
            501,
            777,
            501,
            778,
            500,
            790,
            500,
            791,
            499,
            794,
            499,
            795,
            498,
            798,
            498,
            799,
            497,
            807,
            497,
            808,
            496,
            816,
            496,
            817,
            495,
            818,
            495,
            819,
            494,
            821,
            494,
            822,
            493,
            823,
            493,
            824,
            492,
            826,
            492,
            827,
            491,
            828,
            491,
            829,
            490,
            830,
            490,
            831,
            489,
            833,
            489,
            834,
            488,
            835,
            488,
            836,
            487,
            839,
            487,
            840,
            486,
            844,
            486,
            845,
            485,
            848,
            485,
            849,
            484,
            853,
            484,
            854,
            483,
            858,
            483,
            859,
            482,
            862,
            482,
            863,
            481,
            871,
            481,
            872,
            480,
            884,
            480,
            885,
            479,
            897,
            479,
            898,
            478,
            910,
            478,
            911,
            477,
            922,
            477,
            923,
            476,
            935,
            476,
            936,
            475,
            948,
            475,
            949,
            474,
            961,
            474,
            962,
            473,
            970,
            473,
            971,
            472,
            974,
            472,
            975,
            471,
            979,
            471,
            980,
            470,
            983,
            470,
            984,
            469,
            987,
            469,
            988,
            468,
            991,
            468,
            992,
            467,
            993,
            467,
            994,
            466,
            995,
            466,
            996,
            465,
            997,
            465,
            998,
            464,
            1001,
            464,
            1002,
            463,
            1008,
            463,
            1009,
            462,
            1014,
            462,
            1015,
            461,
            1019,
            461,
            1020,
            460,
            1021,
            460,
            1022,
            459,
            1023,
            459,
            1024,
            458,
            1026,
            458,
            1027,
            457,
            1028,
            457,
            1029,
            456,
            1030,
            456,
            1031,
            455,
            1032,
            455,
            1033,
            454,
            1034,
            454,
            1044,
            444,
            1044,
            439,
            1045,
            438,
            1045,
            428,
            1046,
            427,
            1046,
            418,
            1047,
            417,
            1047,
            411,
            1046,
            410,
            1046,
            407,
            1045,
            406,
            1045,
            404,
            1044,
            403,
            1044,
            400,
            1043,
            399,
            1043,
            396,
            1042,
            395,
            1042,
            393,
            1041,
            392,
            1041,
            389,
            1040,
            388,
            1040,
            387,
            1038,
            385,
            1038,
            384,
            1033,
            379,
            1033,
            378,
            1028,
            373,
            1028,
            372,
            1024,
            368,
            1024,
            367,
            1019,
            362,
            1019,
            361,
            1016,
            358,
            1016,
            357,
            1015,
            356,
            1015,
            355,
            1014,
            354,
            1014,
            353,
            1013,
            352,
            1013,
            351,
            1012,
            350,
            1012,
            349,
            1011,
            348,
            1011,
            347,
            1009,
            345,
            1009,
            344,
            1006,
            341,
            1006,
            340,
            1003,
            337,
            1003,
            336,
            1000,
            333,
            1000,
            332,
            997,
            329,
            997,
            328,
            996,
            327,
            996,
            326,
            995,
            325,
            995,
            323,
            994,
            322,
            994,
            321,
            993,
            320,
            993,
            318,
            992,
            317,
            992,
            315,
            991,
            314,
            991,
            313,
            986,
            308,
            986,
            307,
            985,
            306,
            985,
            305,
            984,
            304,
            984,
            303,
            983,
            302,
            983,
            301,
            982,
            300,
            982,
            299,
            981,
            298,
            981,
            297,
            974,
            290,
            974,
            289,
            973,
            288,
            973,
            287,
            971,
            285,
            971,
            284,
            970,
            283,
            970,
            282,
            969,
            281,
            969,
            280,
            962,
            273,
            962,
            272,
            961,
            271,
            961,
            270,
            960,
            269,
            960,
            268,
            959,
            267,
            959,
            266,
            958,
            265,
            958,
            264,
            957,
            263,
            957,
            262,
            956,
            261,
            956,
            260,
            955,
            259,
            955,
            258,
            954,
            257,
            954,
            256,
            953,
            255,
            953,
            254,
            952,
            253,
            952,
            252,
            951,
            251,
            951,
            250,
            950,
            249,
            950,
            248,
            948,
            246,
            948,
            245,
            944,
            241,
            944,
            240,
            941,
            237,
            941,
            236,
            937,
            232,
            937,
            231,
            926,
            220,
            925,
            220,
            915,
            210,
            914,
            210,
            912,
            208,
            911,
            208,
            909,
            206,
            908,
            206,
            906,
            204,
            905,
            204,
            904,
            203,
            903,
            203,
            901,
            201,
            900,
            201,
            898,
            199,
            897,
            199,
            895,
            197,
            894,
            197,
            893,
            196,
            892,
            196,
            891,
            195,
            890,
            195,
            889,
            194,
            888,
            194,
            887,
            193,
            886,
            193,
            885,
            192,
            884,
            192,
            883,
            191,
            882,
            191,
            881,
            190,
            880,
            190,
            879,
            189,
            878,
            189,
            877,
            188,
            876,
            188,
            874,
            186,
            873,
            186,
            870,
            183,
            869,
            183,
            866,
            180,
            865,
            180,
            864,
            179,
            863,
            179,
            862,
            178,
            861,
            178,
            860,
            177,
            859,
            177,
            858,
            176,
            857,
            176,
            856,
            175,
            855,
            175,
            853,
            173,
            852,
            173,
            851,
            172,
            850,
            172,
            849,
            171,
            848,
            171,
            847,
            170,
            846,
            170,
            845,
            169,
            844,
            169,
            843,
            168,
            842,
            168,
            841,
            167,
            840,
            167,
            838,
            165,
            837,
            165,
            836,
            164,
            835,
            164,
            834,
            163,
            833,
            163,
            832,
            162,
            831,
            162,
            830,
            161,
            829,
            161,
            828,
            160,
            827,
            160,
            825,
            158,
            824,
            158,
            823,
            157,
            822,
            157,
            821,
            156,
            820,
            156,
            819,
            155,
            818,
            155,
            817,
            154,
            816,
            154,
            815,
            153,
            814,
            153,
            813,
            152,
            812,
            152,
            810,
            150,
            809,
            150,
            808,
            149,
            807,
            149,
            806,
            148,
            805,
            148,
            804,
            147,
            803,
            147,
            802,
            146,
            801,
            146,
            800,
            145,
            799,
            145,
            798,
            144,
            797,
            144,
            796,
            143,
            794,
            143,
            793,
            142,
            791,
            142,
            790,
            141,
            788,
            141,
            787,
            140,
            785,
            140,
            784,
            139,
            782,
            139,
            781,
            138,
            779,
            138,
            778,
            137,
            777,
            137,
            776,
            136,
            775,
            136,
            774,
            135,
            773,
            135,
            772,
            134,
            771,
            134,
            770,
            133,
            769,
            133,
            767,
            131,
            766,
            131,
            765,
            130,
            764,
            130,
            763,
            129,
            762,
            129,
            761,
            128,
            760,
            128,
            759,
            127,
            758,
            127,
            757,
            126,
            754,
            126,
            753,
            125,
            750,
            125,
            749,
            124,
            746,
            124,
            745,
            123,
            742,
            123,
            741,
            122,
            738,
            122,
            737,
            121,
            734,
            121,
            733,
            120,
            730,
            120,
            729,
            119,
            727,
            119,
            726,
            118,
            725,
            118,
            724,
            117,
            722,
            117,
            721,
            116,
            719,
            116,
            718,
            115,
            717,
            115,
            716,
            114,
            714,
            114,
            713,
            113,
            711,
            113,
            710,
            112,
            709,
            112,
            708,
            111,
            704,
            111,
            703,
            110,
            698,
            110,
            697,
            109,
            691,
            109,
            690,
            108,
            685,
            108,
            684,
            107
         ],
         "iscrowd": 0
      }
   ]
}
```
image:
![241295174_802164147128543_3685831163417916581_n](https://github.com/roboflow/supervision/assets/31528034/c8283804-ebfd-45f1-9746-e47ec8e6505a)

That's all the information I can provide, hope it helps

**Comment by SkalskiP** [comment]
Looks legit @yance-dev! I'm reopening the issue. Would you like to help us work on the fix?

**Comment by onuralpszr** [comment]
@SkalskiP I would like to take on this looks fairly easy.

**Comment by onuralpszr** [comment]
You can also test with your data and I will test based on this one as well

**Comment by SkalskiP** [comment]
@onuralpszr fell free to take a look! I'm a bit surprised we mutate the structure. 😮

**Comment by yance-dev** [comment]
> @SkalskiP I would like to take on this looks fairly easy. You can also test with your data and I will test based on this one as well

Yes, the affected code may be only one or two lines, so the original maintainer may be able to deal with this issue more quickly.

In addition, due to different time zones, my reply may not be so timely. happy weekend😂

**Comment by SkalskiP** [comment]
@yance-dev, no worries! We will try to understand and solve the problem. Thanks again for bringing this to our attention.

**Comment by mayankagarwals** [comment]
@SkalskiP @yance-dev I was able to reproduce this issue in colab 
https://colab.research.google.com/drive/1E2H8MkgFjMeJznOLUc6npaAQuFkRRz7-?usp=sharing

But want to reproduce it locally to deep dive into why the shape seems to be changing. 
Can someone point me to the sample dataset used in the example notebooks?

**Comment by SkalskiP** [comment]
Hi @mayankagarwals 👋🏻! Yup! Here it is: https://universe.roboflow.com/athanasios-kokkinhs-iqro0/aedespupaeheads

**Comment by SkalskiP** [comment]
Hi, @yance-dev 👋🏻! I hope you are well. It looks like @mayankagarwals submitted PR with the fix for this issue #225. Could you test it to make sure we are correct this time?
